splash-screen/android/src/main/java/com/capacitorjs/plugins/splashscreen/SplashScreen.java

```diff
@@ -45,12 +54,24 @@ public class SplashScreen {
      * @param activity
      */
     public void showOnLaunch(final AppCompatActivity activity) {
-        if (config.getLaunchShowDuration() == 0) {
-            return;
-        }
         SplashScreenSettings settings = new SplashScreenSettings();
         settings.setShowDuration(config.getLaunchShowDuration());
         settings.setAutoHide(config.isLaunchAutoHide());
+
+        // Method can fail if styles are incorrectly set...
+        // If it fails, log error & fallback to old method
+        try {
+            showWithAndroid12API(activity, settings);
+            return;
+        } catch (Exception e) {
+            Logger.warn("Android 12 Splash API failed... using previous method.");
+            this.onPreDrawListener = null;
+        }
+
+        if (config.getLaunchShowDuration() == 0) {
+            return;
+        }
```

Comment:
This should remain on top, otherwise if users are using `"launchShowDuration": 0`, I briefly see the default android icon (green robot) between the splash and the app.